pages/api/notification.ts

```diff
@@ -0,0 +1,35 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import * as webPush from "web-push";
+
+webPush.setVapidDetails(
+    `mailto:${process.env["WEB_PUSH_EMAIL"]}`,
+    process.env["NEXT_PUBLIC_WEB_PUSH_PUBLIC_KEY"] ?? "",
+    process.env["WEB_PUSH_PRIVATE_KEY"] ?? "",
+);
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+    if (req.method !== "POST") {
+        res.status(405).end();
+    }
+    const { subscription, message, webPushPrivateKey } = req.body;
+
+    if (!subscription || !message) {
+        return res.status(400).send("Subscription details and message are required.");
+    }
+
+    if (webPushPrivateKey !== process.env["WEB_PUSH_PRIVATE_KEY"]) {
```

Comment:
Maybe a litt ugly to send the private key like this, but we need to restrict the endpoint. If you have any other suggestions, please let me know. 😄